addons/auth_totp_mail_enforce/tests/test_auth_signup.py

```diff
@@ -0,0 +1,46 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.addons.base.tests.common import HttpCaseWithUserDemo, HttpCaseWithUserPortal
+from odoo.tests import tagged
+
+
+@tagged('post_install', '-at_install', 'assets_bundle')
+class TestAuthSignupFlow(HttpCaseWithUserPortal, HttpCaseWithUserDemo):
+
+    def setUp(self):
+        super().setUp()
+        res_config = self.env['res.config.settings']
+        self.default_values = res_config.default_get(list(res_config.fields_get()))
+
+    def _activate_free_signup(self):
+        self.default_values.update({'auth_signup_uninvited': 'b2c'})
+
+    def _enforce_2fa(self):
+        self.default_values.update({'auth_totp.policy': 'all_required'})
+
+    def test_signup_with_2fa_enforced(self):
+        """
+        Check that registration cleanly succeeds with 2FA enabled and enforced
+        """
+
+        self._activate_free_signup()
+        self._enforce_2fa()
```

Comment:
I would prefer you do what you do in `setUp` here and replace those two function two, with a nice comment. In general, I am not a fan of function that are defined to be called only once, especially when they are one-liners.